backend/src/main/java/harustudy/backend/auth/util/AesTokenProvider.java

```diff
@@ -0,0 +1,83 @@
+package harustudy.backend.auth.util;
+
+import harustudy.backend.auth.exception.InvalidAccessTokenException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Base64;
+import java.util.Date;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AesTokenProvider {
+
+    private static final String alg = "AES/CBC/PKCS5Padding";
+    private static final String iv = "0123456789abcdef"; // 16byte
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("MMddHH:mm:ssyyyy");
+
+    public String createAccessToken(Long subject, Long accessTokenExpireLength, String secretKey) {
+        try {
+            Date now = new Date();
+            Date expireAt = new Date(now.getTime() + accessTokenExpireLength);
+            String formatted = DATE_FORMAT.format(expireAt);
+            String text = subject + " " + formatted;
+            return encrypt(text, secretKey);
+        } catch (Exception e) {
+            throw new InvalidAccessTokenException(e);
+        }
+    }
+
+    private String encrypt(String text, String secretKey) throws GeneralSecurityException {
+        Cipher cipher = Cipher.getInstance(alg);
+        SecretKeySpec keySpec = new SecretKeySpec(secretKey.getBytes(), "AES");
+        IvParameterSpec ivParamSpec = new IvParameterSpec(iv.getBytes());
+        cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivParamSpec);
+
+        byte[] encrypted = cipher.doFinal(text.getBytes(StandardCharsets.UTF_8));
+        return Base64.getEncoder().encodeToString(encrypted);
+    }
+
+    public Long parseSubject(String accessToken, String secretKey) {
+        try {
+            String[] splitted = decrypt(accessToken, secretKey);
+            validateLength(splitted);
+            validateExpiration(splitted);
+            return parseSubject(splitted);
+        } catch (Exception e) {
```

Comment:
여기도 Exception을 catch하고 있군요

Reply:
여기는 uncheckedException과 checkedException 모든 상황에 대한 예외가 다양해서 구체적인 예외를 모두 나열하는 것이 필요가 없을 것 같아서 Exception으로 처리했어요.
예를 들어 ParseException은 CheckedException이고 RuntimeException을 상속한 저희의 커스텀 예외인 InvalidAccessTokenException 및 배열 크기 관련한 ArrayIndexOutOfBoundsException도 발생하고...

특히나 내부에서 InvalidAccessTokenException를 반환하는데 이를 다시 InvalidAccessTokenException으로 전달하는 것은 오히려 어색하지 않나 싶어서요. 또 Exception으로 모아도 잃는 stacktrace 정보도 없으니 괜찮지 않을까 했어요
예외에 대해선 여러 고민하고 하나로 묶은 것인데 그래도 구체적인 예외를 나열하는 편이 낫다고 생각하시나요?

Reply:
```suggestion
        } catch (ParseException |
                 GeneralSecurityException|
                 RuntimeException e) {
            throw new InvalidAccessTokenException(e);
        }
```

조금 번거롭더라도 이 코드를 나중에 보게될 저희 스스로를 위해서 catch문을 분리하는 것도 하나의 방법이 될 수 있지 않을까 제안해봅니다. 

그게 아니라면 개인적으로 private 검증 메서드마다 발생하는 checkedException에 대해서 예외 전환을 각각 해주는 편이 좋다고 생각이 들었어요. encrypt메서드의 경우는 이미 내부에서 예외를 런타임 예외로 전환해주고 있어서 크게 걸릴 부분은 없을 것 같은데 어떻게 생각하시나요?

Reply:
해당 부분은 별도 클래스에서 처리하도록 해서 예외들을 깔끔하게 처리했습니다!

추가로 확인해주세요~!